src/CalcViewModel/Common/TraceLogger.cpp

```diff
@@ -236,782 +121,215 @@ namespace CalculatorApp
-            }
-        }
-        // update the functionCount with total function count which we are tracking through tracelog.
-        functionCount = i;
-    }
-
-    wstring TraceLogger::GetProgrammerType(int index)
-    {
-        if (index >= 0)
-        {
-            return s_programmerType[index];
-        }
-        // return "N/A"
-        return s_programmerType[0];
+        fields.AddUInt64(PDT_PRIVACY_DATA_TAG, PDT_PRODUCT_AND_SERVICE_USAGE);
+        LogLevel2Event(EVENT_NAME_EXCEPTION, fields);
     }
 
-    bool TraceLogger::GetIndex(int& index)
+    void TraceLogger::UpdateButtonUsage(int buttonId, ViewMode mode)
     {
-        if (findIndex[index] > 0)
+        // IsProgrammerMode, IsScientificMode, IsStandardMode and None are not actual buttons, so ignore them
+        if (buttonId == (int)NumbersAndOperatorsEnum::IsProgrammerMode || buttonId == (int)NumbersAndOperatorsEnum::IsScientificMode ||
+            buttonId == (int)NumbersAndOperatorsEnum::IsStandardMode || buttonId == (int)NumbersAndOperatorsEnum::None)
         {
-            index = findIndex[index];
-            return true;
-        }
-        return false;
-    }
-
-    void TraceLogger::UpdateWindowCount(size_t windowCount)
-    {
-        maxWindowCount = (maxWindowCount > windowCount) ? maxWindowCount : windowCount;
-        windowLaunchCount++;
-    }
-
-    void TraceLogger::LogMaxWindowCount()
-    {
-        if (!GetTraceLoggingProviderEnabled())
-            return;
-
-        LoggingFields fields{};
-        fields.AddUInt32(L"WindowCount", (unsigned int)maxWindowCount);
-        LogTelemetryEvent(EVENT_NAME_MAX_WINDOW_COUNT, fields);
-    }
-
-    void TraceLogger::LogWindowActivated() const
-    {
-        if (!GetTraceLoggingProviderEnabled())
             return;
-
-        LoggingFields fields{};
-        LogTelemetryEvent(EVENT_NAME_WINDOW_LAUNCHED_PROTOCOL, fields);
-    }
-
-    void TraceLogger::LogWindowLaunched() const
-    {
-        if (!GetTraceLoggingProviderEnabled())
-            return;
-
-        LoggingFields fields{};
-        LogTelemetryEvent(EVENT_NAME_WINDOW_LAUNCHED_TILESEARCH, fields);
-    }
-
-    void TraceLogger::LogBitLengthButtonUsed(int windowId)
-    {
-        if (bitLengthButtonUsage.find(windowId) == bitLengthButtonUsage.end())
-        {
-            bitLengthButtonUsage.insert(pair<int, int>(windowId, 1));
         }
-        else
-        {
-            bitLengthButtonUsage[windowId]++;
-        }
-        if ((bitLengthButtonUsage[windowId] == 5) && !bitLengthButtonLoggedInSession)
-        {
-            LoggingFields fields{};
-            fields.AddUInt32(L"WindowId", windowId);
-            LogTelemetryEvent(EVENT_NAME_BITLENGTH_BUTTON_USED, fields);
-
-            bitLengthButtonLoggedInSession = true;
-        }
-    }
 
-    void TraceLogger::LogRadixButtonUsed(int windowId)
-    {
-        if (radixButtonUsage.find(windowId) == radixButtonUsage.end())
+        // Writer lock for the static resources
```

Comment:
>Writer lock for the static resources [](start = 11, length = 36)

Comment should probably be updated here and on line 106--windowIdLog and buttonLog are not "static resources".